app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -377,10 +376,6 @@ private void showDialog(final List<AdapterChoiceItem> choices) {
         final boolean isExtAudioEnabled = preferences.getBoolean(
                 getString(R.string.use_external_audio_player_key), false);
 
-        returnList.add(new AdapterChoiceItem(getString(R.string.show_info_key),
-                getString(R.string.show_info),
-                resolveResourceIdFromAttr(context, R.attr.ic_info_outline)));
-
         if (capabilities.contains(VIDEO) && !(isExtVideoEnabled && linkType != LinkType.STREAM)) {
             returnList.add(new AdapterChoiceItem(getString(R.string.video_player_key),
                     getString(R.string.video_player),
```

Comment:
In order to achieve what was discussed in #4071, I'd just add a condition here that shows "R.string.show_info" if autoplay is off, but still uses "R.string.video_player_key" as the key in any case. At the end of the day both options open the same activity, even though the outcome is different, so I think users would expect "Show info" to be selected in the dialog if they had used the "Video player" choice before disabling autoplay, and viceversa. 

I don't think you would need to do anything else in order to achieve what I described above, since VideoDetailFragment already handles autoplay by itself.